src/Core/src/Handlers/WebView/WebViewHandler.Android.cs

```diff
@@ -1,40 +1,79 @@
 ﻿using Android.Webkit;
-using Android.Widget;
 using static Android.Views.ViewGroup;
 using AWebView = Android.Webkit.WebView;
 
 namespace Microsoft.Maui.Handlers
 {
 	public partial class WebViewHandler : ViewHandler<IWebView, AWebView>
 	{
+		WebViewClient? _webViewClient;
+		WebChromeClient? _webChromeClient;
+		bool _settingsSet;
+
 		protected override AWebView CreateNativeView()
 		{
-			var aWebView = new MauiWebView(Context!)
+			return new MauiWebView(Context!)
 			{
-#pragma warning disable 618 // This can probably be replaced with LinearLayout(LayoutParams.MatchParent, LayoutParams.MatchParent); just need to test that theory
-				LayoutParameters = new AbsoluteLayout.LayoutParams(LayoutParams.MatchParent, LayoutParams.MatchParent, 0, 0)
-#pragma warning restore 618
+				LayoutParameters = new LayoutParams(LayoutParams.MatchParent, LayoutParams.WrapContent)
 			};
-
-			if (aWebView.Settings != null)
-			{
-				aWebView.Settings.JavaScriptEnabled = true;
-				aWebView.Settings.DomStorageEnabled = true;
-			}
-			return aWebView;
 		}
 
 		protected override void DisconnectHandler(AWebView nativeView)
 		{
 			nativeView.StopLoading();
-			nativeView?.Dispose();
 		}
 
 		public static void MapSource(WebViewHandler handler, IWebView webView)
 		{
 			IWebViewDelegate? webViewDelegate = handler.NativeView as IWebViewDelegate;
 
+			SetSettingsAndClient(handler);
+
 			handler.NativeView?.UpdateSource(webView, webViewDelegate);
 		}
+
+		public static void MapWebViewClient(WebViewHandler handler, IWebView webView)
+		{
+			if (handler._webViewClient == null)
+			{
+				handler._webViewClient = new WebViewClient();
+				handler.NativeView.SetWebViewClient(handler._webViewClient);
+			}
+		}
+
+		public static void MapWebChromeClient(WebViewHandler handler, IWebView webView)
+		{
+			if (handler._webChromeClient == null)
+			{
+				handler._webChromeClient = new WebChromeClient();
+				handler.NativeView.SetWebChromeClient(handler._webChromeClient);
+			}
+		}
+
+		public static void MapWebViewSettings(WebViewHandler handler, IWebView webView)
+		{
+			if (!handler._settingsSet)
+			{
+				handler.NativeView.UpdateSettings(webView, true, true);
+				handler._settingsSet = true;
+			}
+		}
+
+		static void SetSettingsAndClient(WebViewHandler handler)
+		{
+			if (handler._webViewClient == null)
```

Comment:
This feels kind of the same as before. I was thinking a bool for each property and then once all of them are true you process the source.

Unless we want to go the route of triggering all three of these anytime the source changes? 


So all of these mappers would call a method `ProcessSourceWhenReady()`

In there once the necessary set of mappers have ran then it'll continue on

Reply:
This feels kind of the same as before. I was thinking a bool for each property and then once all of them are true you process the source.

Unless we want to go the route of triggering all three of these anytime the source changes?

So all of these mappers would call a method ProcessSourceWhenReady()

In there once the necessary set of mappers have ran then it'll continue on